src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -228,15 +203,12 @@ public GroupShardsIterator allAssignedShardsGrouped(String[] indices, boolean in
     public GroupShardsIterator allAssignedShardsGrouped(String[] indices, boolean includeEmpty, boolean includeRelocationTargets) throws IndexMissingException {
         // use list here since we need to maintain identity across shards
         ArrayList<ShardIterator> set = new ArrayList<>();
-        if (indices == null || indices.length == 0) {
-            indices = indicesRouting.keySet().toArray(new String[indicesRouting.keySet().size()]);
-        }
         for (String index : indices) {
             IndexRoutingTable indexRoutingTable = index(index);
             if (indexRoutingTable == null) {
                 continue;
                 // we simply ignore indices that don't exists (make sense for operations that use it currently)
-//                throw new IndexMissingException(new Index(index));
+                // throw new IndexMissingException(new Index(index));
```

Comment:
what happened here? we just want to leave this commented out? maybe we want to even remove this line then?